modules/core/test/test_io.cpp

```diff
@@ -1345,3 +1345,24 @@ TEST(Core_InputOutput, FileStorage_json_bool)
     ASSERT_EQ((int)fs["bool_false"], 0);
     fs.release();
 }
+
+TEST(Core_InputOutput, FileStorage_free_file_after_exception)
+{
+    const std::string fileName = "test.yml";
+    const std::string content = "%YAML:1.0\n cameraMatrix;:: !<tag:yaml.org,2002:opencv-matrix>\n";
+
+    fstream testFile;
+    testFile.open(fileName.c_str(), std::fstream::out);
+    if(!testFile.is_open()) FAIL();
+    testFile << content;
+    testFile.close();
+
+    try
+    {
+        FileStorage fs(fileName, FileStorage::READ + FileStorage::FORMAT_YAML);
+    }
```

Comment:
I would add an assertion failure after the constructor, e.g. assert that ``!fs.isOpened()`` to be sure that future changes (i.e. not throwing an exception) don't render this test useless.

Similarly, I would move the ``std::remove`` to after the catch block and leave the block empty